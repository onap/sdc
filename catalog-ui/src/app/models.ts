/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by ob0695 on 2/23/2017.
 */
import from = require("core-js/fn/array/from");
export * from './models/activity';
export * from './models/additional-information';
export * from './models/app-config';
export * from './models/validation-config';
export * from './models/plugins-config';
export * from './models/artifacts';
export * from './models/aschema-property';
export * from './models/schema-attribute';
export * from './models/attributes';
export * from './models/capability';
export * from './models/category';
export * from './models/comments';
export * from './models/components/component';
export * from './models/components/displayComponent';
export * from './models/components/resource';
export * from './models/components/service';
export * from './models/componentsInstances/componentInstance';
export * from './models/componentsInstances/resourceInstance';
export * from './models/componentsInstances/serviceInstance';
export * from './models/componentsInstances/serviceProxyInstance';
export * from './models/componentsInstances/serviceSubstitutionInstance';
export * from './models/graph/zones/group-instance';
export * from './models/graph/zones/policy-instance';
export * from './models/graph/zones/zone';
export * from './models/graph/zones/zone-instance';
export * from './models/csar-component';
//export * from './models/data-type-properties';
export * from './models/properties-inputs/property-be-model';
export * from './models/properties-inputs/property-fe-model';
export * from './models/properties-inputs/property-fe-map';
export * from './models/properties-inputs/derived-fe-property';
export * from './models/properties-inputs/property-declare-api-model';
export * from './models/properties-inputs/property-input-detail';
export * from './models/properties-inputs/input-fe-model';
export * from './models/properties-inputs/simple-flat-property';
export * from './models/data-types-map';
export * from './models/data-types';
export * from './models/distribution';
export * from './models/export-excel';
export * from './models/file-download';
export * from './models/graph/graph-links/common-base-link';
export * from './models/graph/graph-links/common-ci-link-base';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-link-base';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-simple-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-ucpe-host-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-ucpe-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-vl-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-vl-ucpe-link';
export * from './models/graph/graph-links/links-factory';
export * from './models/graph/graph-links/module-graph-links/module-ci-link-base';
export * from './models/graph/graph-links/module-graph-links/module-ci-vl-link';
export * from './models/graph/graphTooltip';
export * from './models/graph/assetPopoverObj';
export * from './models/graph/link-menu';
export * from './models/graph/match-relation';
export * from './models/graph/nodes/base-common-node';
export * from './models/graph/nodes/common-ci-node-base';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-base';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-cp';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-configuration';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service-proxy';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service-substitution';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-ucpe-cp';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-ucpe';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vf';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vfc';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vl';
export * from './models/graph/nodes/modules-graph-nodes/module-node-base';
export * from './models/graph/nodes/nodes-factory';
export * from './models/graph/point';
export * from './models/graph/connectRelationModel';
export * from './models/graph/relationship';
export * from './models/heat-parameters';
export * from './models/input-property-base';
export * from './models/inputs-and-properties';
export * from './models/inputs';
export * from './models/instance-inputs-properties-map';
export * from './models/left-panel';
export * from './models/member';
export * from './models/modules/base-module';
export * from './models/operation';
export * from './models/properties';
export * from './models/requirement';
export * from './models/server-error-response';
export * from './models/tab';
export * from './models/tooltip-data';
export * from './models/user';
export * from './models/validate';
export * from './models/component-metadata';
export * from './models/components-array';
export * from './models/modal';
export * from './models/button';
export * from './models/wizard-step';
export * from './models/radio-button';
export * from './models/filter-properties-assignment-data';
export * from './models/properties-inputs/input-be-model';
export * from './models/catalogSelector';
export * from './models/componentsInstances/fullComponentInstance';
export * from './models/catalogSelector';
export * from './models/componentsInstances/fullComponentInstance';
export * from './models/service-instance-properties-and-interfaces';
export * from './models/relationship-types';
export * from './models/tosca-presentation';
export * from './models/node-types';
export * from './models/capability-types';