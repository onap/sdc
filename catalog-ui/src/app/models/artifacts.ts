/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

'use strict';

import {ArtifactType} from './../utils';
import {HeatParameterModel} from "./heat-parameters";

//this object contains keys, each key contain ArtifactModel
export class ArtifactGroupModel {
    
    constructor(artifacts?:ArtifactGroupModel) {
        _.forEach(artifacts, (artifact:ArtifactModel, key) => {
            this[key] = new ArtifactModel(artifact);
        });
    }

    public filteredByType(type:string):ArtifactGroupModel {
        let tmpArtifactGroupModel = new ArtifactGroupModel();
        _.each(Object.keys(this), (key)=>{
            if (this[key].artifactType === type) {
                tmpArtifactGroupModel[key] = this[key];
            }
        });
        return tmpArtifactGroupModel;
    };
}

export class ArtifactModel {

    artifactDisplayName:string;
    artifactGroupType:string;
    uniqueId:string;
    artifactName:string;
    artifactLabel:string;
    artifactType:string;
    artifactUUID:string;
    artifactVersion:string;
    creatorFullName:string;
    creationDate:number;
    lastUpdateDate:number;
    description:string;
    mandatory:boolean;
    serviceApi:boolean;
    payloadData:string;
    timeout:number;
    esId:string;
    "Content-MD5":string;
    artifactChecksum:string;
    apiUrl:string;
    heatParameters:Array<HeatParameterModel>;
    generatedFromId:string;

    //custom properties
    selected:boolean;
    originalDescription:string;
    envArtifact:ArtifactModel;

    constructor(artifact?:ArtifactModel) {
        if (artifact) {
            this.artifactDisplayName = artifact.artifactDisplayName;
            this.artifactGroupType = artifact.artifactGroupType;
            this.uniqueId = artifact.uniqueId;
            this.artifactName = artifact.artifactName;
            this.artifactLabel = artifact.artifactLabel;
            this.artifactType = artifact.artifactType;
            this.artifactUUID = artifact.artifactUUID;
            this.artifactVersion = artifact.artifactVersion;
            this.creatorFullName = artifact.creatorFullName;
            this.creationDate = artifact.creationDate;
            this.lastUpdateDate = artifact.lastUpdateDate;
            this.description = artifact.description;
            this.mandatory = artifact.mandatory;
            this.serviceApi = artifact.serviceApi;
            this.payloadData = artifact.payloadData;
            this.timeout = artifact.timeout;
            this.esId = artifact.esId;
            this["Content-MD5"] = artifact["Content-MD5"];
            this.artifactChecksum = artifact.artifactChecksum;
            this.apiUrl = artifact.apiUrl;
            this.heatParameters = _.sortBy(artifact.heatParameters, 'name');
            this.generatedFromId = artifact.generatedFromId;
            this.selected = artifact.selected ? artifact.selected : false;
            this.originalDescription = artifact.description;
        }
    }

    public isHEAT = ():boolean => {
        return ArtifactType.HEAT === this.artifactType || ArtifactType.HEAT_VOL === this.artifactType || ArtifactType.HEAT_NET === this.artifactType;
    };

    public isThirdParty = ():boolean => {
        return _.has(ArtifactType.THIRD_PARTY_RESERVED_TYPES, this.artifactType);
    };

    public toJSON = ():any => {
        this.selected = undefined;
        this.originalDescription = undefined;
        this.envArtifact = undefined;
        return this;
    };
}


