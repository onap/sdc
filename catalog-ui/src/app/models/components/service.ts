/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by obarda on 2/4/2016.
 */
'use strict';
import * as _ from "lodash";
import {IServiceService} from "../../services/components/service-service";
import {Component, PropertyModel, DisplayModule, InputsAndProperties, InputModel, InstancesInputsOrPropertiesMapData, InstancesInputsPropertiesMap,
    Distribution, DistributionComponent, ArtifactGroupModel} from "../../models";
import {ArtifactGroupType} from "../../utils/constants";
import {ComponentMetadata} from "../component-metadata";
import {ForwardingPath} from "app/models/forwarding-path";

export class Service extends Component {

    public serviceApiArtifacts:ArtifactGroupModel;
    public componentService:IServiceService;
    public ecompGeneratedNaming:boolean;
    public namingPolicy:string;
    public serviceType:string;
    public serviceRole:string;
    public serviceFunction:string;
    public environmentContext:string;
    public instantiationType:string;
    public forwardingPaths:{ [key:string]:ForwardingPath } = {};

    constructor(componentService:IServiceService, $q:ng.IQService, component?:Service) {
        super(componentService, $q, component);
        this.ecompGeneratedNaming = true;
        if (component) {
            this.serviceApiArtifacts = new ArtifactGroupModel(component.serviceApiArtifacts);
            this.filterTerm = this.name + ' ' + this.description + ' ' + (this.tags ? this.tags.toString() : '') + ' ' + this.version;
            this.ecompGeneratedNaming = component.ecompGeneratedNaming;
            this.namingPolicy = component.namingPolicy;
            this.serviceType = component.serviceType;
            this.serviceRole = component.serviceRole;
            this.serviceFunction = component.serviceFunction;
            this.instantiationType = component.instantiationType;
            this.environmentContext = component.environmentContext;
            if (component.categories && component.categories[0]) {
                this.mainCategory = component.categories[0].name;
                this.selectedCategory = this.mainCategory;
            }
        }
        this.componentService = componentService;
        this.iconSprite = "sprite-services-icons";
    }

    public getDistributionsList = ():ng.IPromise<Array<Distribution>> => {
        return this.componentService.getDistributionsList(this.uuid);
    };

    public getDistributionsComponent = (distributionId:string):ng.IPromise<Array<DistributionComponent>> => {
        return this.componentService.getDistributionComponents(distributionId);
    };

    public markAsDeployed = (distributionId:string):ng.IPromise<any> => {
        return this.componentService.markAsDeployed(this.uniqueId, distributionId);
    };

    /* we need to change the name of the input to vfInstanceName + input name before sending to server in order to create the inputs on the service
     *  we also need to remove already selected inputs (the inputs that already create on server, and disabled in the view - but they are selected so they are still in the view model
     */
    public createInputsFormInstances = (instancesInputsMap:InstancesInputsOrPropertiesMapData, instancePropertiesMap:InstancesInputsOrPropertiesMapData):ng.IPromise<Array<InputModel>> => {

        let deferred = this.$q.defer<Array<InputModel>>();
        let onSuccess = (inputsCreated:Array<InputModel>):void => {
            this.inputs = inputsCreated.concat(this.inputs);
            deferred.resolve(inputsCreated);
        };
        let onFailed = (error:any):void => {
            deferred.reject(error);
        };

        let propertiesAndInputsMap:InstancesInputsPropertiesMap = new InstancesInputsPropertiesMap(instancesInputsMap, instancePropertiesMap);
        propertiesAndInputsMap = propertiesAndInputsMap.cleanUnnecessaryDataBeforeSending(); // We need to create a copy of the map, without the already selected inputs / properties, and to send the clean map
        this.componentService.createInputsFromInstancesInputs(this.uniqueId, propertiesAndInputsMap).then(onSuccess, onFailed);
        return deferred.promise;
    };

    // we need to change the name of the input to vfInstanceName + input name before sending to server in order to create the inputs on the service
    public getServiceInputInputsAndProperties = (inputId:string):ng.IPromise<InputsAndProperties> => {
        let deferred = this.$q.defer<InputsAndProperties>();
        let onSuccess = (inputsAndProperties:InputsAndProperties):void => {
            let input:InputModel = _.find(this.inputs, (input:InputModel) => {
                return input.uniqueId === inputId;
            });
            input.inputs = inputsAndProperties.inputs;
            input.properties = inputsAndProperties.properties;
            deferred.resolve(inputsAndProperties);
        };
        let onFailed = (error:any):void => {
            deferred.reject(error);
        };
        this.componentService.getComponentInputInputsAndProperties(this.uniqueId, inputId).then(onSuccess, onFailed);
        return deferred.promise;
    };

    public deleteServiceInput = (inputId:string):ng.IPromise<InputModel> => {
        let deferred = this.$q.defer<InputModel>();

        let onSuccess = (deletedInput:InputModel):void => {
            delete _.remove(this.inputs, {uniqueId: deletedInput.uniqueId})[0];
            deferred.resolve(deletedInput);
        };

        let onFailed = (error:any):void => {
            deferred.reject(error);
        };

        this.componentService.deleteComponentInput(this.uniqueId, inputId).then(onSuccess, onFailed);
        return deferred.promise;
    };

    public getArtifactsByType = (artifactGroupType:string):ArtifactGroupModel => {
        switch (artifactGroupType) {
            case ArtifactGroupType.DEPLOYMENT:
                return this.deploymentArtifacts;
            case ArtifactGroupType.INFORMATION:
                return this.artifacts;
            case ArtifactGroupType.SERVICE_API:
                return this.serviceApiArtifacts;
        }
    };

    public updateGroupInstanceProperties = (resourceInstanceId:string, group:DisplayModule, properties:Array<PropertyModel>):ng.IPromise<Array<PropertyModel>> => {

        let deferred = this.$q.defer<Array<PropertyModel>>();
        let onSuccess = (updatedProperties:Array<PropertyModel>):void => {
            _.forEach(updatedProperties, (property:PropertyModel) => { // Replace all updated properties on the we needed to update
                _.extend(_.find(group.properties, {uniqueId: property.uniqueId}), property);
            });
            deferred.resolve(updatedProperties);
        };
        let onError = (error:any):void => {
            deferred.reject(error);
        };

        this.componentService.updateGroupInstanceProperties(this.uniqueId, resourceInstanceId, group.groupInstanceUniqueId, properties).then(onSuccess, onError);
        return deferred.promise;
    };

    getTypeUrl():string {
        return 'services/';
    }


    public setComponentMetadata(componentMetadata: ComponentMetadata) {
        super.setComponentMetadata(componentMetadata);
        this.ecompGeneratedNaming = componentMetadata.ecompGeneratedNaming;
        this.namingPolicy = componentMetadata.namingPolicy;
        this.serviceType = componentMetadata.serviceType;
        this.serviceRole = componentMetadata.serviceRole;
        this.serviceFunction = componentMetadata.serviceFunction;
        this.environmentContext = componentMetadata.environmentContext;
        this.instantiationType = componentMetadata.instantiationType;
        this.setComponentDisplayData();
    }

    setComponentDisplayData():void {
        this.filterTerm = this.name + ' ' + this.description + ' ' + (this.tags ? this.tags.toString() : '') + ' ' + this.version;
        if (this.categories && this.categories[0]) {
            this.mainCategory = this.categories[0].name;
            this.selectedCategory = this.mainCategory;
        }
        this.iconSprite = "sprite-services-icons";
    }

    public toJSON = ():any => {
        let temp = angular.copy(this);
        temp.componentService = undefined;
        temp.filterTerm = undefined;
        temp.iconSprite = undefined;
        temp.mainCategory = undefined;
        temp.subCategory = undefined;
        temp.selectedInstance = undefined;
        temp.showMenu = undefined;
        temp.$q = undefined;
        temp.selectedCategory = undefined;
        temp.modules = undefined;
        temp.groupInstances = undefined;
        temp.policies = undefined;
        return temp;
    };
}

