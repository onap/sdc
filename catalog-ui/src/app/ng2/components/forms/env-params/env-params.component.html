<div class="filter-bar">
    <sdc-filter-bar
            [placeHolder]="'Search...'"
            [testId]="'search-env-param-name'"
            (keyup)="updateFilter($event)">
    </sdc-filter-bar>
</div>

<ngx-datatable
        class='material'
        [rows]='artifact.heatParameters'
        [columnMode]="'flex'"
        [headerHeight]="40"
        [rowHeight]="'auto'"
        [scrollbarV]="false">

    <ngx-datatable-column name="Parameter" [flexGrow]="2">
        <ng-template let-row="row" ngx-datatable-cell-template prop="name">
            {{row.name}}
            <span *ngIf="row.description.length > 0" class="info">
                <svg-icon [name]="'comment'" (click)="openPopOver('',row.description,{x:$event.pageX , y:$event.pageY },'bottom')"></svg-icon>
            </span>
        </ng-template>
    </ngx-datatable-column>

    <ngx-datatable-column name="DefaultValue"[flexGrow]="1">
        <ng-template let-row="row" let-value="value" ngx-datatable-cell-template>
            {{row.defaultValue}}
        </ng-template>
    </ngx-datatable-column>

    <ngx-datatable-column name="CurrentValue" [flexGrow]="2">
        <ng-template let-row="row" let-value="value" ngx-datatable-cell-template>
            <sdc-input class="sdc-input-wrapper" #numberValidator
                [placeHolder]="'Enter text'"
                [isViewMode]="isViewOnly"
                [size]="'medium'"
                [(value)]=row.currentValue
                [isIconClickable]="true"
                (onRighIconClicked)="clearCurrentValue(row.name)"
                [righIconName]="'trash-o'"
                [testId] = "'value-field-of-' + row.name">
            </sdc-input>

            <sdc-validation [validateElement]="numberValidator" (validityChanged)="onValidityChange($event)" [disabled]="false" [testId]="_testId">
                <sdc-regex-validator *ngIf="displayRegexValid && row.type == 'number' && row.currentValue !== null" [message]="'Value should be of type number.'" [pattern]="displayRegexValid" [disabled]="false"></sdc-regex-validator>
            </sdc-validation>
        </ng-template>
    </ngx-datatable-column>

</ngx-datatable>

<div *ngIf="isInstanceSelected" class="artifactTimeout">

    <sdc-number-input
               label="Deployment Timeout ({{minDeploymentTimeout}}-{{maxDeploymentTimeout}} minutes)"
               [required]="true"
               [disabled]="false"
               name="artifactTimeout"
               testId="deploymentTimeout"
               value="{{artifact.timeout}}"
               [maxValue]="maxDeploymentTimeout"
               [minValue]="minDeploymentTimeout"
               (valueChange)="timeoutChanged($event)"
               [isViewMode]="isViewOnly"
               [step]="1"
               >

    </sdc-number-input>

</div>