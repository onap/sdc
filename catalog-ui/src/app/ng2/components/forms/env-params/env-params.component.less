.filter-bar {
  padding-bottom: 25px;
}

:host ::ng-deep {

  .ngx-datatable {
    //border: 1px solid red;
    .datatable-body-cell {
      .info {
        float: right;
      }
    }
  }
}

.artifactTimeout{
  padding-top: 25px;
  justify-content: start;
  width: 230px;
}
