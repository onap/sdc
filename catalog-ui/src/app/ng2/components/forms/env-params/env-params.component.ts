/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
import { SdcUiCommon, SdcUiServices } from 'onap-ui-angular';
import { Subject } from 'rxjs/Rx';
import { ArtifactModel } from '../../../../models/artifacts';
import { CacheService } from '../../../services/cache.service';

export interface IPoint {
    x: number;
    y: number;
}

@Component({
    selector: 'env-params',
    templateUrl: './env-params.component.html',
    styleUrls: ['../../../../../assets/styles/table-style.less', './env-params.component.less']
})
export class EnvParamsComponent implements OnInit {

    @Input() public artifact: ArtifactModel;
    @Input() public isInstanceSelected: boolean;
    @Input() public isViewOnly: boolean;

    @ViewChild('textArea') textArea: ElementRef;
    private copiedWorkingArtifactHeatParameters = [];
    private onValidationChange: Subject<boolean> = new Subject();
    private displayRegexValid = SdcUiCommon.RegexPatterns.numberOrEmpty;

    // Deployment timeout in minutes
    private maxDeploymentTimeout: number = 120;
    private minDeploymentTimeout: number = 1;
    private defaultDeploymentTimeout: number = 60;

    constructor(private cacheService: CacheService, private popoverService: SdcUiServices.PopoverService) {
        const configuration = cacheService.get('UIConfiguration');
        if (configuration && configuration.heatDeploymentTimeout) {
            this.maxDeploymentTimeout = configuration.heatDeploymentTimeout.maxMinutes;
            this.minDeploymentTimeout = configuration.heatDeploymentTimeout.minMinutes;
            this.defaultDeploymentTimeout = configuration.heatDeploymentTimeout.defaultMinutes;
        }
    }

    ngOnInit(): void {
        this.copiedWorkingArtifactHeatParameters = [...this.artifact.heatParameters];
    }

    public clearCurrentValue = (name: string) => {
        this.artifact.heatParameters.filter((param) => param.name === name)[0].currentValue = '';
    }

    public timeoutChanged(timeout) {
        this.artifact.timeout = timeout;
    }

    updateFilter(event) {
        const val = event.target.value.toLowerCase();
        // filter our data
        const temp = this.copiedWorkingArtifactHeatParameters.filter((param) => {
            return !val || param.name ? param.name.toLowerCase().indexOf(val) !== -1 : -1 || param.currentValue ? param.currentValue.toLowerCase().indexOf(val) !== -1 : -1;
        });
        // update the rows
        this.artifact.heatParameters = temp;
    }

    private openPopOver = (title: string, content: string, positionOnPage: IPoint, location: string) => {
        this.popoverService.createPopOver(title, content, positionOnPage, location);
    }

    private onValidityChange = (isValid: boolean): void => {
        this.onValidationChange.next(isValid);
    }
}
