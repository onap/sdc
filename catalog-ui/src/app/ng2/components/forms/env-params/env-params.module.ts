import { NgModule } from "@angular/core";
import { EnvParamsComponent } from "./env-params.component";
import { NgxDatatableModule } from "@swimlane/ngx-datatable";
import { SdcUiComponentsModule, SdcUiServices } from "onap-ui-angular";


@NgModule({
    declarations: [
        EnvParamsComponent
    ],
    imports: [
        NgxDatatableModule,
        SdcUiComponentsModule
    ],
    exports: [
        EnvParamsComponent
    ],
    entryComponents: [ //need to add anything that will be dynamically created
        EnvParamsComponent
    ],
    providers: [
        SdcUiServices.ModalService
    ]
})

export class EnvParamsModule {

}
