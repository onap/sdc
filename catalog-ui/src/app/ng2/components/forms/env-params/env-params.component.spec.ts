import {async, ComponentFixture} from "@angular/core/testing";
import {EnvParamsComponent} from "./env-params.component";
import {SdcUiServices, SdcUiCommon} from "onap-ui-angular";
import {ConfigureFn, configureTests} from "../../../../../jest/test-config.helper";
import {NgxDatatableModule} from "@swimlane/ngx-datatable";
import {NO_ERRORS_SCHEMA} from "@angular/core";
import {ArtifactModel} from "../../../../models/artifacts";
import { CacheService } from '../../../services/cache.service';

describe('environment parameters component', () => {

    let fixture: ComponentFixture<EnvParamsComponent>;
    let popoverServiceMock: Partial<SdcUiServices.PopoverService>;
    let regexPatterns: any;

    let artifactModel = new ArtifactModel();

    let mockHeatParameters = [
        {currentValue: "1", defaultValue: null, description: "Description 1", empty:false, name: "Param1", ownerId: null, type: "string", uniqueId: null, envDisplayName:null, version: null, filterTerm:null},
        {currentValue: "2", defaultValue: null, description: "Description 2", empty:false, name: "Param2", ownerId: null, type: "string", uniqueId: null, envDisplayName:null, version: null, filterTerm:null},
        {currentValue: "3", defaultValue: null, description: "Description 3", empty:false, name: "Param3", ownerId: null, type: "string", uniqueId: null, envDisplayName:null, version: null, filterTerm:null}
        ];

    let keyboardEvent = new KeyboardEvent("keyup");

    beforeEach(
        async(() => {

            popoverServiceMock = {
                createPopOver : jest.fn()
            }

            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [EnvParamsComponent],
                    imports: [NgxDatatableModule],
                    schemas: [NO_ERRORS_SCHEMA],
                    providers: [
                        { provide: SdcUiServices.PopoverService, useValue: popoverServiceMock },
                        { provide: CacheService, useValue: { get: jest.fn() } }
                    ],
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(EnvParamsComponent);
            });
        })
    );

    it('should match current snapshot of env-params component', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('should clear CurrentValue for a given name in the heat parameter', () => {
         fixture.componentInstance.artifact = artifactModel;
         fixture.componentInstance.artifact.heatParameters = mockHeatParameters;
         expect(fixture.componentInstance.artifact.heatParameters.length).toBe(3);
         expect(fixture.componentInstance.artifact.heatParameters[0].currentValue).toBe("1");
         fixture.componentInstance.clearCurrentValue("Param1");
         expect(fixture.componentInstance.artifact.heatParameters[0].currentValue).toBe("");
    });

    it("should update filter heatParameters so there won''t be any value to be displayed", () => {

        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.artifact.heatParameters = mockHeatParameters;
        fixture.componentInstance.ngOnInit();

        let event = {
             target : {
                value: 'paramNotExist'
             }
        }

        expect(fixture.componentInstance.artifact.heatParameters.length).toBe(3);
        fixture.componentInstance.updateFilter(event);
        expect(fixture.componentInstance.artifact.heatParameters.length).toBe(0);
    });

    it("should update filter heatParameters so there will be only 1 value to be displayed", () => {

        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.artifact.heatParameters = mockHeatParameters;
        fixture.componentInstance.ngOnInit();

        let event = {
            target : {
                value: 'param1'
            }
        }

        expect(fixture.componentInstance.artifact.heatParameters.length).toBe(3);
        fixture.componentInstance.updateFilter(event);
        expect(fixture.componentInstance.artifact.heatParameters.length).toBe(1);
    });

});