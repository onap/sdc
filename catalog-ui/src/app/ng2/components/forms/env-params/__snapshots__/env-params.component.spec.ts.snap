// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`environment parameters component should match current snapshot of env-params component 1`] = `
<env-params
  cacheService={[Function Object]}
  clearCurrentValue={[Function Function]}
  copiedWorkingArtifactHeatParameters={[Function Array]}
  defaultDeploymentTimeout={[Function Number]}
  displayRegexValid={[Function String]}
  maxDeploymentTimeout={[Function Number]}
  minDeploymentTimeout={[Function Number]}
  onValidationChange={[Function Subject]}
  onValidityChange={[Function Function]}
  openPopOver={[Function Function]}
  popoverService={[Function Object]}
  textArea="undefined"
>
  <div
    class="filter-bar"
  >
    <sdc-filter-bar />
  </div><ngx-datatable
    class="material ngx-datatable"
  >
    <div
      visibilityobserver=""
    >
      
      <datatable-body
        class="datatable-body"
      >
        <datatable-selection>
          
          
          
        </datatable-selection>
      </datatable-body>
      
    </div>
  </ngx-datatable>
</env-params>
`;
